### ISSUE TITLE
Question about the ent_loss/Hloss in codes.

### ISSUE THREAD

**Issue by Zhengyang1995** [Xavier-Pan/WSGCN#2]
Hey, thank you for your amazing work. It is really interesting!
I have a question about the ent_loss. In the paper, as shown in equation ( 5 ), the ent_loss is the mean of  **log(q) * q** . 
But in the function named HLoss in /GCN/utils.py , it is **torch.exp(x) * x**.
Why it is not **torch.log(x) * x**?
Thank you for your time~

**Comment by ghost** [comment]
Hi, I have the same question. Do you figure out why using torch.exp(x) * x? Have you tried the torch.log(x) * x in the experiment?

**Comment by johnnylu305** [comment]
Hi, @Zhengyangnuaa  and @GraceLi66. Sorry, I did not get the issue/message from github. If you check the return of the GCN, it is F.log_softmax(x, dim=1). That is why we use torch.exp(x) * x. I hope this can answer your question.

**Comment by Zhengyang1995** [comment]
Cool, so does it work better for log_softmax than softmax,

**Comment by johnnylu305** [comment]
I think it is just a training/implementation trick that is numerical stable. If you put it into the loss function, the equation is still the same. Therefore, I do not think there would be a huge difference if you use softmax with torch.log(x) * x.
